shell/platform/linux/fl_view.cc

```diff
@@ -175,31 +40,32 @@ static gboolean fl_view_send_pointer_button_event(FlView* self,
       return FALSE;
   }
   int old_button_state = self->button_state;
+  FlutterPointerPhase phase;
   if (event->type == GDK_BUTTON_PRESS) {
     // Drop the event if Flutter already thinks the button is down
     if ((self->button_state & button) != 0)
       return FALSE;
     self->button_state ^= button;
 
-    fl_event.phase = old_button_state == 0 ? kDown : kMove;
+    phase = old_button_state == 0 ? kDown : kMove;
   } else if (event->type == GDK_BUTTON_RELEASE) {
     // Drop the event if Flutter already thinks the button is up
     if ((self->button_state & button) == 0)
       return FALSE;
     self->button_state ^= button;
 
-    fl_event.phase = self->button_state == 0 ? kUp : kMove;
+    phase = self->button_state == 0 ? kUp : kMove;
   }
 
-  if (self->flutter_engine == nullptr)
+  if (self->engine == nullptr)
     return FALSE;
 
-  fl_event.x = event->x;
-  fl_event.y = event->y;
-  fl_event.device_kind = kFlutterPointerDeviceKindMouse;
-  fl_event.buttons = self->button_state;
-  FlutterEngineSendPointerEvent(self->flutter_engine, &fl_event, 1);
-
+  fl_engine_send_mouse_pointer_event(self->engine,
+                                       phase,
+                                       event->time * 1000,
```

Comment:
Nit: let's used a named constant for the 1000 so it's obvious from local context what units it's converting from/to.